objects/obj_bomb_select/Draw_0.gml

```diff
@@ -16,71 +14,137 @@ if (max_ships>0)and (instance_exists(obj_star_select)){
     
     draw_set_halign(fa_left);
     draw_set_font(fnt_menu);
-    if (all_sel=0){draw_set_alpha(0.5);draw_text(xx+76,yy+81,string_hash_to_newline("[ ]"));}
-    if (all_sel=1){draw_set_alpha(1);draw_text(xx+76,yy+81,string_hash_to_newline("[X]"));}
+    if (all_sel==0){
+        draw_set_alpha(0.5);
+        draw_text(xx+76,yy+81,string_hash_to_newline("[ ]"));
+    }
+    if (all_sel==1){
+        draw_set_alpha(1);
+        draw_text(xx+76,yy+81,string_hash_to_newline("[X]"));
+    }
     draw_set_alpha(1);
     draw_set_font(fnt_info);
     
     draw_text(xx+310,yy+218,string_hash_to_newline("Current Selection:"));
     draw_text(xx+310.5,yy+218.5,string_hash_to_newline("Current Selection:"));
     
-    
-    var hers;
-    var influ;
-    var poppy;
+    var hers, influ, poppy;
+
     hers=p_target.p_heresy[obj_controller.selecting_planet]+p_target.p_heresy_secret[obj_controller.selecting_planet];
     influ=p_target.p_influence[obj_controller.selecting_planet];
-    if (p_target.p_large[obj_controller.selecting_planet]=1) then poppy=string(p_target.p_population[obj_controller.selecting_planet])+"B";
-    if (p_target.p_large[obj_controller.selecting_planet]=0) then poppy=string(scr_display_number(p_target.p_population[obj_controller.selecting_planet]));
-    
-    
-    
+    if (p_target.p_large[obj_controller.selecting_planet]==1) then poppy=string(p_target.p_population[obj_controller.selecting_planet])+"B";
+    if (p_target.p_large[obj_controller.selecting_planet]==0) then poppy=string(scr_display_number(p_target.p_population[obj_controller.selecting_planet]));
     
     draw_text(xx+310,yy+283,string_hash_to_newline("Pop.: "+string(poppy)));
     draw_text(xx+310.5,yy+283.5,string_hash_to_newline("Pop.:"));
     
-    
-    
     if (p_target.sprite_index!=spr_star_hulk){
         draw_text(xx+310,yy+317,string_hash_to_newline("Target: "));
         draw_text(xx+310,yy+317.5,string_hash_to_newline("Target: "));
         
         draw_set_halign(fa_center);
         
-        var t_name;t_name="";
-        if (target=2) then t_name="Imperial Forces";
-        if (target=2.5) and (p_target.p_owner[obj_controller.selecting_planet]=8) then t_name="Gue'la Forces";
-        if (target=3) then t_name="Mechanicus";
-        if (target=5) then t_name="Ecclesiarchy";
-        if (target=6) then t_name="Eldar";
-        if (target=7) then t_name="Orks";
-        if (target=8) then t_name="Tau";
-        if (target=9) then t_name="Tyranids";
-        if (target=10) then t_name="Chaos";
-        if (target=13) then t_name="Necrons";
+        var t_name = "";
+        switch (target) {
```

Comment:
TODO a centralised point to be able to fetch display names from factions identifying number